test/test_overrides.py

```diff
@@ -0,0 +1,756 @@
+import torch
+import numpy as np
+import unittest
+import six
+import inspect
+import pprint
+
+from common_utils import TestCase
+
+HANDLED_FUNCTIONS = {}
+HANDLED_FUNCTIONS_SUB = {}
+HANDLED_FUNCTIONS_SUB_DIAGONAL = {}
+HANDLED_FUNCTIONS_TENSOR_LIKE = {}
+
+def implements(torch_function):
+    "Register an implementation of a torch function for a Tensor-like object."
+    def decorator(func):
+        HANDLED_FUNCTIONS[torch_function.__name__] = func
+        return func
+    return decorator
+
+def implements_sub(torch_function):
+    "Register an implementation of a torch function for a Tensor-like object."
```

Comment:
The doc string of this function is exactly the same as the one above. But clearly it does something different. What is it doing differently? (I am sure I can work it out once I finish reading, but the docs here are not helping me...)